app/src/util/itertools.ts

```diff
@@ -84,6 +100,46 @@ function* filter<T>(predicate: (x: T) => boolean, input: Generator<T>): Generato
   }
 }
 
+function* dropwhile<T>(predicate: (x: T) => boolean, input: Generator<T>): Generator<T> {
+  for (const item of input) {
+    if (!predicate(item)) {
+      yield item;
+      break;
+    }
+  }
+  yield* input;
+}
+
+function* dropuntil<T>(predicate: (x: T) => boolean, input: Generator<T>): Generator<T> {
+  for (const item of input) {
+    if (predicate(item)) {
+      yield item;
```

Comment:
this seems wrong too?

Reply:
This drops elements from the beginning of the generator until the predicate is met for the first time

Reply:
ok got it